src/main/java/com/google/api/codegen/discovery/config/go/GoTypeNameGenerator.java

```diff
@@ -29,6 +30,20 @@
   private static final Pattern SUB_VERSION = Pattern.compile("^(.+)_(v[0-9.]+)$");
 
   @Override
+  public List<String> getMethodNameComponents(List<String> nameComponents) {
+    // Don't edit the original object.
+    LinkedList<String> copy = new LinkedList<String>(nameComponents);
+    copy.removeFirst();
```

Comment:
Is this what the comment two lines above is referring to? If so, move it down one line. And maybe you could expand on what "original object" means.